dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/domain/batching/DataProcessorFactory.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.domain.batching
+
+import com.datadog.android.core.internal.data.file.FileOrchestrator
+import com.datadog.android.core.internal.data.file.ImmediateFileWriter
+import com.datadog.android.core.internal.domain.FilePersistenceConfig
+import com.datadog.android.core.internal.domain.Serializer
+import com.datadog.android.core.internal.domain.batching.processors.DataProcessor
+import com.datadog.android.core.internal.domain.batching.processors.DefaultDataProcessor
+import com.datadog.android.core.internal.domain.batching.processors.NoOpDataProcessor
+import com.datadog.android.privacy.TrackingConsent
+import java.io.File
+import java.util.concurrent.ExecutorService
+
+internal class DataProcessorFactory<T : Any>(
+    private val intermediaryFolderPath: String,
+    private val targetFolderPath: String,
+    private val filePersistenceConfig: FilePersistenceConfig,
+    private val serializer: Serializer<T>,
+    private val executorService: ExecutorService
+) {
+
+    fun resolveProcessor(consent: TrackingConsent): DataProcessor<T> {
+        return when (consent) {
+            TrackingConsent.PENDING -> {
+                DefaultDataProcessor(
+                    executorService,
+                    ImmediateFileWriter(buildFileOrchestrator(intermediaryFolderPath), serializer)
+                )
```

Comment:
How would we return a different `DataProcessor<T>` from here? Will it be possible? - I don't clearly see how we could do it in Kotlin.

Reply:
You're thinking ahead :). It is not possible for now but for this first part we will not need that as we will only have a DP implementation which writes in a folder. Later on when we will introduce the Sampling part we will indeed need something specific for RUM which writes in session folders. In that case we might probably need a DataProcessor with a specific FileWriter but I will introduce the change once we get there.